pandas/io/tests/test_pickle.py

```diff
@@ -284,6 +284,46 @@ def test_pickle_v0_15_2(self):
         #
         tm.assert_categorical_equal(cat, pd.read_pickle(pickle_path))
 
+    def compression_explicit(self, compression):
+        # issue 11666
+        if compression == 'xz':
+            tm._skip_if_no_lzma()
+        with tm.ensure_clean(self.path) as path:
+            df = tm.makeDataFrame()
+            df.to_pickle(path, compression=compression)
+            df2 = pd.read_pickle(path, compression=compression)
```

Comment:
> Using command line maybe better but involves more environment configurations. 

I was only suggesting using your local command line "to add some prepared .pickle.xz file into pandas/io/tests/data".

> reading these files may fail as the pickle version varies.

What about using a pickle with protocol version 2? [`pickle.load`](https://docs.python.org/3/library/pickle.html#pickle.load) should automatically detect the version 2, which is supported in 2.7 and 3.